web/src/click_handlers.ts

```diff
@@ -983,6 +983,33 @@ export function initialize(): void {
         }
     });
 
+    const compose_buttons = tippy.default(".compose_button_tooltip", {
+        trigger: "mouseenter",
+        placement: "top",
+    });
+
+    tippy.createSingleton(compose_buttons, {
+        onTrigger(instance, event) {
+            const currentTarget = event.currentTarget;
+            if (currentTarget instanceof HTMLElement) {
+                const templateId = currentTarget.dataset.tooltipTemplateId;
+                if (templateId) {
+                    const templateElement = document.querySelector(`#${templateId}`);
+                    if (templateElement) {
+                        const parsedHTML = ui_util.parse_html(templateElement.innerHTML);
+                        instance.setContent(parsedHTML);
+                    }
+                }
+
+                if (currentTarget.classList?.contains("disabled-on-hover")) {
+                    instance.setProps({delay: 0});
+                } else {
+                    instance.setProps({delay: 500});
+                }
+            }
+        },
+    });
+
```

Comment:
This code doesn't look it belongs in `click_handlers.ts`. If I jog down my memory since that last time I worked on singletons, does it have to do with the instances showing up as empty when used in other places? Say `compose_tooltips.ts`?

Reply:
If so, we should still consider defining this logic in `compose_tooltips.ts` and instead initialize the singleton along with it's child instances when the compose box is opened or when the message edit UI is opened.